frontend/src/components/home-page/components/filter-dialog.tsx

```diff
@@ -0,0 +1,38 @@
+import { useState } from 'hooks/hooks';
```

Comment:
each component should be in the separate page

Reply:
Take this filter-dialog component and that menu-button from above and make it into a single component that can be controlled via props. Put it in the global components/common directory and use it in your page.

Reply:
![image](https://user-images.githubusercontent.com/43746781/127656567-d0cb9105-87a4-4d9c-ba67-1a0d9666cc34.png)
![image](https://user-images.githubusercontent.com/43746781/127656636-76b76ac0-cc0b-4ed2-8195-ad0b988dd84d.png)
Please tell me, am I on the right track? I started making a generic component that includes a menu button and a filter dialog.

Reply:
I mean a generic component should have 2 kinds of property types and 2 functions MenuButton and FilterDialog?